src/V3EmitMkJson.cpp

```diff
@@ -0,0 +1,320 @@
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << "\"" << name << "\": " << type << "\n";
+            m_prefix += m_indent;
+            m_scope.push(type == '{' ? '}' : ']');
+            m_empty = true;
+            return *this;
+        }
+
+        Printer& put(const std::string& name, const std::string& value) {
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << "\"" << name << "\": \"" << value << "\"";
+            m_empty = false;
+            return *this;
+        }
+
+        Printer& put(const std::string& name, bool value) {
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << "\"" << name << "\": " << (value ? "true" : "false");
+            m_empty = false;
+            return *this;
+        }
+
+        Printer& put(const std::string& name, int value) {
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << "\"" << name << "\": " << value;
+            m_empty = false;
+            return *this;
+        }
+
+        Printer& begin(char type = '{') {
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << type << "\n";
+            m_prefix += m_indent;
+            m_scope.push(type == '{' ? '}' : ']');
+            m_empty = true;
+            return *this;
+        }
+
+        Printer& put(const std::string& value) {
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << "\"" << value << "\"";
+            m_empty = false;
+            return *this;
+        }
+
+        Printer& put(bool value) {
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << (value ? "true" : "false");
+            m_empty = false;
+            return *this;
+        }
+
+        Printer& put(int value) {
+            if (!m_empty) { *m_of << ",\n"; }
+            *m_of << m_prefix << value;
+            m_empty = false;
+            return *this;
+        }
+
+        template <typename T>
+        Printer& putList(const std::string& name, const T& list) {
+            if (list.empty()) { return *this; }
+            begin(name, '[');
+            for (auto it = list.begin(); it != list.end(); ++it) { put(*it); }
+            return end();
+        }
+
+        Printer& end() {
+            assert(m_prefix.length() >= m_indent.length());
+            m_prefix.erase(m_prefix.end() - m_indent.length(), m_prefix.end());
+            assert(!m_scope.empty());
+            *m_of << "\n" << m_prefix << m_scope.top();
+            m_scope.pop();
+            return *this;
+        }
+
+        Printer& operator+=(Printer& cursor) {
+            // meaningless syntax sugar, at least for now
+            return *this;
+        }
+    };
+
+    // METHODS
+
+    // STATIC FUNCTIONS
+
+    static string universalAbsolutePath(const string& filename) {
+        return V3Os::fileSlashPath(V3Os::filenameRealPath(filename));
+    }
```

Comment:
Call target function directly.

Reply:
That makes code unreadable. Too many V30s:: and so on.